modules/AOR_Reports/AOR_Report.php

```diff
@@ -1097,32 +1121,68 @@ function build_report_query_select($query = array(), $group_value =''){
-                        $query['second_group_by'][] = $select_field;
-                    } else {
-                        $query['group_by'][] = $select_field;
-                    }
-                } elseif ($field->field_function != null) {
-                    $select_field = $field->field_function . '(' . $select_field . ')';
-                } else {
-                    $query['second_group_by'][] = $select_field;
+                if($field->group_by == 1) {
+                    $has_group_by = true;
+                    $query['group_by'][] = $field->format ? str_replace('(%1)', '(' . $select_field . ')', preg_replace(array('/\s+/', '/Y/', '/m/', '/d/'), array(', ', 'YEAR(%1)', 'MONTH(%1)', 'DAY(%1)'), trim(preg_replace('/[^Ymd]/', ' ', $field->format)))) : $select_field;
+                }
+                else {
+                    $query['second_group_by'][] = $field->format ? str_replace('(%1)', '(' . $select_field . ')', preg_replace(array('/\s+/', '/Y/', '/m/', '/d/'), array(', ', 'YEAR(%1)', 'MONTH(%1)', 'DAY(%1)'), trim(preg_replace('/[^Ymd]/', ' ', $field->format)))) : $select_field;
+                }
+
+                if($field->field_function != null){
+                    $select_field = $field->field_function.'('.$select_field.')';
                 }
 
                 if($field->sort_by != ''){
                     $query['sort_by'][] = $select_field." ".$field->sort_by;
                 }
 
+
+
                 $query['select'][] = $select_field ." AS '".$field->label."'";
 
-                if($field->group_display == 1 && $group_value) $query['where'][] = $select_field." = '".$group_value."' AND ";
-                    ++$i;
+                if($field->group_display == 1 && $group_value) {
+                    $query['where'][] = $select_field." = '".$group_value."' AND ";
+                }
+
+                ++$i;
+            }
+        }
+
+        if($has_group_by) {
+            // add fields to the group by MS SQL Requirement
+            foreach($query['select_id'] as $s => $select) {
+                $query['group_by'][] = $select['field'];
             }
         }
+
+        $this->injectParentModuleIDField($module->table_name, $query);
+
         return $query;
     }
 
+    /**
+     * Injects the parent/root table id into the query['select'] key.
+     *
+     * Ensures that all fields which display as hyperlinks have an id/record to link to
+     *
+     * @param $table_name - (string) table name
+     * @param $query - (array) used to build up query
+     */
+    function injectParentModuleIDField ($table_name, &$query) {
+        $needle = $this->db->quoteIdentifier($table_name).".id";
+        $found = false;
+
+        // find if the key already exists
+        foreach($query['select'] as $select) {
+            if(!empty(stristr($select, $needle))) { $found = true; break; }
```

Comment:
I get another write context error: PHP Fatal error:  Can't use function return value in write context.
Change the function:
if(stristr($select, $needle) !== FALSE) { $found = true; break; }